_ml-commons-plugin/agents-tools/mcp/mcp-connector.md

```diff
@@ -0,0 +1,254 @@
+
+For example, to register an OpenAI chat model, send the following request:
+
+```json
+POST /_plugins/_ml/models/_register
+{
+  "name": "My OpenAI model: gpt-4",
+  "function_name": "remote",
+  "description": "Test model registration (this example uses OpenAI, but you can register any model)",
+  "connector": {
+    "name": "My OpenAI Connector: gpt-4",
+    "description": "Connector for the OpenAI chat model",
+    "version": 1,
+    "protocol": "http",
+    "parameters": {
+      "model": "gpt-4o"
+    },
+    "credential": {
+      "openAI_key": "<YOUR_API_KEY>"
+    },
+    "actions": [
+      {
+        "action_type": "predict",
+        "method": "POST",
+        "url": "https://api.openai.com/v1/chat/completions",
+        "headers": {
+          "Authorization": "Bearer ${credential.openAI_key}"
+        },
+        "request_body": "{ \"model\": \"${parameters.model}\", \"messages\": [{\"role\":\"developer\",\"content\":\"${parameters.system_instruction}\"},${parameters._chat_history:-}{\"role\":\"user\",\"content\":\"${parameters.prompt}\"}${parameters._interactions:-}], \"tools\": [${parameters._tools:-}],\"parallel_tool_calls\":${parameters.parallel_tool_calls},\"tool_choice\": \"${parameters.tool_choice}\" }"
+      }
+    ]
+  }
+}
+```
+{% include copy-curl.html %}
+
+The response contains the model ID:
+
+```json
+{
+  "task_id": "K_iQfpYBjoQOEoSHN3wU",
+  "status": "CREATED",
+  "model_id": "LPiQfpYBjoQOEoSHN3zH"
+}
+```
+
+To check the status of the operation, provide the task ID to the [Tasks API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/tasks-apis/get-task/). Once the registration is complete, the task `state` changes to `COMPLETED`.
+
+## Step 3: Register an agent for accessing MCP tools
+
+Currently, MCP tools can only be used with [_conversational_ agents]({{site.url}}{{site.baseurl}}/ml-commons-plugin/agents-tools/index/#conversational-agents) or _plan, execute, and reflect_ agent types. 
+<!-- TODO: add a link to the agent type when 9743 is merged -->
+
+To enable external MCP tools, include one or more MCP connectors in your agent's configuration.
+
+Each connector must specify the following parameters in the `parameters.mcp_connectors` array.
+
+| Parameter | Data type | Required | Description | 
+|:--- |:--- |:--- |:--- |
+| `mcp_connector_id` | String | Yes | The connector ID of the MCP connector. | 
+| `tool_filters` | Array | No | An array of Java-style regular expressions that specify which tools from the MCP server to make available to the agent. If omitted or set to an empty array, all tools exposed by the connector will be available. Use `^/$` anchors or literal strings to precisely match tool names. For example, `^get_forecast` matches any tool starting with get_forecast, while `search_indices` matches only `search_indices`.|
```

Comment:
Should "get_forecast" be in quotation marks, as below?

Reply:
Yes!